server/queries/db.js

```diff
@@ -17,23 +18,98 @@ sequelize.sync({ alter: true }).then(() => {
     console.log("Database connected.");
 });
 
-module.exports.createUser = username => {
+module.exports.validatePassword = (username, password) => {
+    return new Promise((resolve, reject) => {
+        User.findOne({
+            where: {
+                username: username
+            }
+        })
+        .then(user => {
+            if (user) {
+                bcrypt.compare(password, user.dataValues.password, (err, res) => {                    
+                    if (res) {
+                        resolve(user);
+                    } else {
+                        reject({
+                            code: 404,
+                            message: "Password is incorrect"
+                        });
+                    }
+                });
+            } else {
+                reject({
+                    code: 404,
+                    message: "User with this username does not exist"
+                });
+            }
+        })
+        .catch(err => {
+            reject({
+                code: 404,
+                message: err.message
+            });
+        });
+    });
+}
+
+module.exports.createUser = (username, password) => {
+    return new Promise((resolve, reject) => {
+        bcrypt.hash(password, 10, (err, hash) => {
+            if (err) {
+                reject({
+                    code: 404,
+                    message: err.message
+                });
+            }
+            User.findOrCreate({
+                where: {
+                    username: username
+                },
+                defaults: { // set the default properties if it doesn't exist
+                    username: username.trim(),
+                    password: hash,
+                    score: 0
+                }
+            })
+            .then(([user, created]) => {                
+                if (!created) {
```

Comment:
this code would read better if you inverted the condition -- otherwise we get a double-negative when we do the else